docs/contributing/our-backward-compatibility-promise.md

```diff
@@ -0,0 +1,71 @@
+# Our Backward Compatibility Promise
+
+Smooth and safe upgrades of your own e-commerce project are very important to us.
+In the same time, we need to be able to improve Shopsys Framework for you by adding functionality, enhancing or simplifying current functions and fixing bugs.
+This is why we've written this page explaining what amount of backward compatibility should be expected and how we plan to make changes in the future.
+
+Shopsys Framework is built on the shoulders of giants so we've based our BC promise on the [Symfony Backward Compatibility Promise](https://symfony.com/doc/3.4/contributing/code/bc.html).
```

Comment:
Please move this paragraph under `Clarifications and Exceptions`.

The flow of the article will be then
* motivation
* versions
* BC promise itself

(currently the flow is not so smooth)

Once you move this paragraph, please rename title `Clarifications and Exceptions` to `BC Promise in Detail`

Reply:
Originally, I wanted to mention the extension of Symfony's BC promise in the intro as the page contains a lot of useful general information about the motivation.